### PR TITLE
[Python API] move ngraph python api to the new location

### PR DESCRIPTION
**Pull request by akuporos**
### Details:
 - Move ngraph Python API from 
 ngraph/python -> runtime/bindings/python/src/compatibility

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: runtime/bindings/python/src/openvino/opset1/__init__.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+from ngraph.opset1.ops import convert_like
+from ngraph.opset1.ops import convolution
+from ngraph.opset1.ops import convolution_backprop_data
+from ngraph.opset1.ops import cos
+from ngraph.opset1.ops import cosh
+from ngraph.opset1.ops import ctc_greedy_decoder
+from ngraph.opset1.ops import deformable_convolution
+from ngraph.opset1.ops import deformable_psroi_pooling
+from ngraph.opset1.ops import depth_to_space
+from ngraph.opset1.ops import detection_output
```
Comment (ilya-lavrenov) [inline runtime/bindings/python/src/openvino/opset1/__init__.py:29]: @ilyachur we have different naming style for IE and ngraph python API in new API

@akuporos should we align it?

Comment (ilya-lavrenov) [reply runtime/bindings/python/src/openvino/opset1/__init__.py:29]: See https://www.python.org/dev/peps/pep-0008/

Comment (akuporos) [reply runtime/bindings/python/src/openvino/opset1/__init__.py]: I decided to add the new graph api in another PR.
Will be fixed there.

File: runtime/bindings/python/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+
+add_subdirectory(thirdparty/pybind11 EXCLUDE_FROM_ALL)
+add_subdirectory("src/compatibility")
```
Comment (ilya-lavrenov) [inline runtime/bindings/python/CMakeLists.txt:7]: "" are not needed

File: runtime/bindings/python/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+if(NGRAPH_UNIT_TEST_ENABLE)
+    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/tests/mock/mock_py_ngraph_frontend)
+    add_dependencies(_pyngraph mock_py_ngraph_frontend)
+    set_target_properties(mock_py_ngraph_frontend PROPERTIES
+            LIBRARY_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_DIRECTORY_BIN}
```
Comment (ilya-lavrenov) [inline runtime/bindings/python/CMakeLists.txt:13]: `LIBRARY_OUTPUT_DIRECTORY_BIN` is not defined here

File: runtime/bindings/python/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+            ARCHIVE_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_DIRECTORY_BIN}
+            COMPILE_PDB_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_DIRECTORY_BIN}
+            PDB_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_DIRECTORY_BIN})
+
+    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/tests/mock/pyngraph_fe_mock_api)
```
Comment (ilya-lavrenov) [inline runtime/bindings/python/CMakeLists.txt:18]: `${CMAKE_CURRENT_SOURCE_DIR}/` can be removed

File: runtime/bindings/python/CMakeLists.txt (hunk 2 above)
Comment (akuporos) [reply runtime/bindings/python/CMakeLists.txt]: fixed

File: runtime/bindings/python/CMakeLists.txt (hunk 4 above)
Comment (akuporos) [reply runtime/bindings/python/CMakeLists.txt]: fixed

File: runtime/bindings/python/CMakeLists.txt (hunk 3 above)
Comment (akuporos) [reply runtime/bindings/python/CMakeLists.txt:14]: defined

**Review by jiwaszki — COMMENTED** [review]
It looks good to me, but there is a need to resolve conflicts.

What flag is going to control which API has to be build? Is for that PR left as it is with `-DENABLE_PYTHON` and actual changes will come with later PRs?

**Comment by akuporos** [comment]
> It looks good to me, but there is a need to resolve conflicts.
> 
> What flag is going to control which API has to be build? Is for that PR left as it is with `-DENABLE_PYTHON` and actual changes will come with later PRs?

Conflicts are already resolved :)
Yes, for this PR it is still `-DENABLE_PYTHON`.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by jiwaszki — APPROVED** [review]

**Comment by tomdol** [comment]
Can we merge https://github.com/openvinotoolkit/openvino/pull/7170 first? It will be very easy to merge master into this PR afterwards. If they are merged in the opposite order I will have a lot of conflicts.

**Comment by tomdol** [comment]
Can we merge https://github.com/openvinotoolkit/openvino/pull/7170 first? It will be very easy to merge master into this PR afterwards. If they are merged in the opposite order I will have a lot of conflicts.

**Comment by akuporos** [comment]
> Can we merge #7170 first? It will be very easy to merge master into this PR afterwards. If they are merged in the opposite order I will have a lot of conflicts.

Sure. When are you going to merge your PR?

**Comment by azhogov** [comment]
CI passed:
https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/231
https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/233

**Pull request merged**